build/images/training-operator/Dockerfile

```diff
@@ -13,11 +13,13 @@ RUN go mod download
 COPY . .
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -a -o manager cmd/training-operator.v1/main.go
+USER root
+RUN CGO_ENABLED=1 GOOS=linux GO111MODULE=on go build -tags strictfipsruntime -a -o manager cmd/training-operator.v1/main.go
```

Comment:
why is CGO_ENABLED set to on in this change ?

Reply:
I think it enables the utilization of share lib on the system when required. Without it the [check-payload](https://github.com/openshift/check-payload) util will not pass.

Reply:
Right, by default Go uses its own crypto module from its standard library, which fails FIPS. With CGO, it links to the system library, e.g. OpenSSL.